hca/dss/__init__.py

```diff
@@ -92,16 +93,20 @@ def download(self, bundle_uuid, replica, version="", dest_name="",
             file_uuid = file_["uuid"]
             file_version = file_["version"]
             filename = file_.get("name", file_uuid)
+            walking_dir = dest_name
 
             globs = metadata_files if file_['indexed'] else data_files
             if not any(fnmatchcase(filename, glob) for glob in globs):
                 continue
 
-            if not os.path.isdir(dest_name):
-                os.makedirs(dest_name)
+            intermediate_path, filename = os.path.split(filename)
+            if intermediate_path:
+                walking_dir = os.path.join(walking_dir, intermediate_path)
+            if not os.path.isdir(walking_dir):
+                os.makedirs(walking_dir)
 
             logger.info("File %s: Retrieving...", filename)
```

Comment:
I think this should log the full filename, including the path, as it is in the bundle.
I think this line:
```
 intermediate_path, filename = os.path.split(filename)
```
should be changed to something like:
```
 intermediate_path, filebasename = os.path.split(filename)
```
Rather than having `filename` representing the full path at one point in the code, then being changed to just the basename at some point, I think it would be better to introduce a new name.
If this is done, the the original `filename` (with the path, if present) would be the correct value to log.

Reply:
implimented

Reply:
done